### PR TITLE
DAOS-6960 test: Do not mount dfuse in /tmp for NLT testing.

### PR DESCRIPTION
**Pull request by ashleypittman**
Create a subdirectory to put dfuse mounts in, so that if there
is a deadlock the post section can read /tmp/*.log without itself
deadlocking if dfuse is unresponsive.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Review by brianjmurrell — APPROVED** [review]
`/tmp/` still feels ripe for hitting a deadlock by some naïve test [writer] trawling through it to find something.  IMHO, `/mnt/` would be more ideal.  Having that set up for the test to use can be done easily enough in [ci/provisioning/post_provision_config_nodes.sh](https://github.com/daos-stack/daos/blob/b391e7f5d28295bba2c27155f011b98b3f96af99/ci/provisioning/post_provision_config_nodes.sh)

**Comment by ashleypittman** [comment]
> `/tmp/` still feels ripe for hitting a deadlock by some naïve test [writer] trawling through it to find something. IMHO, `/mnt/` would be more ideal. Having that set up for the test to use can be done easily enough in [ci/provisioning/post_provision_config_nodes.sh](https://github.com/daos-stack/daos/blob/b391e7f5d28295bba2c27155f011b98b3f96af99/ci/provisioning/post_provision_config_nodes.sh)

This was just easier than adding a command line option to NLT.  Looking at that script could I put it in /localhome/jenkins/?

**Comment by brianjmurrell** [comment]
> Looking at that script could I put it in /localhome/jenkins/?

I would think so.

File: ci/unit/test_nlt_node.sh [hunk 1]
```diff
@@ -20,4 +20,4 @@ cd build
 sudo bash -c ". ./utils/sl/setup_local.sh; ./utils/setup_daos_admin.sh"
 
 # NLT will mount /mnt/daos itself.
-./utils/node_local_test.py --max-log-size 400MiB --perf-check all
+./utils/node_local_test.py --max-log-size 400MiB --dfuse-dir /localhome/jenkins/ --perf-check all
```
Comment (brianjmurrell) [inline ci/unit/test_nlt_node.sh:23]: Not sure if the 80-chars warning is still relevant, but an easy fix for that if it is:
```suggestion
./utils/node_local_test.py --max-log-size 400MiB --dfuse-dir /localhome/jenkins/ \
                           --perf-check all
```

**Review by brianjmurrell — APPROVED** [review]
Looks good.  I like the defensiveness of this now.

I just want to confirm though, my last review was a +1, I hope, as that was my intention.  If I didn't +1 it was entirely in error.

**Comment by ashleypittman** [comment]
> Looks good. I like the defensiveness of this now.
> 
> I just want to confirm though, my last review was a +1, I hope, as that was my intention. If I didn't +1 it was entirely in error.

I had missed that, but I didn't realise when I wrote it how easy it would be to add.

I'll fix the shell warning another time.

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**